core/src/main/java/com/linecorp/armeria/internal/Http1ObjectEncoder.java

```diff
@@ -311,7 +308,7 @@ private void convert(int streamId, HttpHeaders inHeaders,
                          boolean isRequest, boolean endStream,
                          HttpHeaders additionalHeaders, HttpHeaders additionalTrailers,
                          boolean enableServerHeader, boolean enableDateHeader,
-                         @Nullable String authorityHeader)
+                         @Nullable SessionProtocol sessionProtocol)
```

Comment:
How about splitting this method into four:

- `convertClientHeaders()` and `convertClientTrailers()`, which don't require `SessionProtocol`
- `convertServerHeaders()` and `convertServerTrailers()`, which require non-null `SessionProtocol`

.. and remove all the `if`s?